core/src/main/scala/com/banno/kafka/package.scala

```diff
@@ -179,30 +120,133 @@ package object kafka {
         x.value == y.value
     }
 
-  implicit object ProducerRecordBifunctor extends Bifunctor[ProducerRecord] {
-    def bimap[A, B, C, D](
+  implicit object ProducerRecordBitraverse extends Bitraverse[ProducerRecord] {
+    override def bifoldLeft[A, B, C](fab: ProducerRecord[A, B], c: C)(
+        f: (C, A) => C,
+        g: (C, B) => C,
+    ): C =
+      g(f(c, fab.key), fab.value)
```

Comment:
I am confused as to how this ensures that we are folding left here. I see no collection, so I am not sure how this folds/collects anything. Regardless, `f()` and `g()` both have the same return type: `C`. So type-wise, I could have easily done `f(c, fab.key)` and everything would still compile just fine. I think the non-descript type parameter names, and the non-descript `f()` and `g()` functions trip me up here.

Reply:
For posterity: answered this on a huddle.